auth_signup_partner_company/models/res_users.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from ast import literal_eval
+
+from odoo import models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    def _signup_create_user(self, values):
+        template_user_id = literal_eval(
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("base.template_portal_user_id", "False")
+        )
+        template_user = self.browse(template_user_id)
+        if not (self.env.company.id in template_user.company_ids.ids):
+            template_user.company_ids = [(4, self.env.company.id)]
+        user = super()._signup_create_user(values)
+        user.company_id = self.env.company
+        user.partner_id.company_id = self.env.company
+        return user
```

Comment:
```suggestion
    def _create_user_from_template(self, values):
        values["company_ids"] = [self.env.company.id]
        values["company_id"] = self.env.company.id
        return super()._create_user_from_template(values)

    def _signup_create_user(self, values):
        user = super()._signup_create_user(values)
        user.partner_id.company_id = self.env.company
        return user
```
Can you try and see if this works fine.